tests/test_federation.py

```diff
@@ -276,3 +276,111 @@ def test_cross_signing_keys_retry(self):
             "ed25519:" + remote_self_signing_key in self_signing_key["keys"].keys(),
         )
         self.assertTrue(remote_self_signing_key in self_signing_key["keys"].values())
+
+
+class StripUnsignedFromEventsTestCase(MessageAcceptTests):
+    def setUp(self):
+        super().setUp()
+
+    def strip_unauthorized_unsigned_values(self):
+        most_recent = self.get_success(
+            self.homeserver.get_datastore().get_latest_event_ids_in_room(self.room_id)
+        )[0]
+        federation_event_handler = self.homeserver.get_federation_event_handler()
+        event1 = make_event_from_dict(
+            {
+                "room_id": self.room_id,
+                "sender": "@baduser:test.serv",
+                "state_key": "@baduser:test.serv",
+                "event_id": "$event1:test.serv",
+                "depth": 1000,
+                "origin_server_ts": 1,
+                "type": "m.room.member",
+                "origin": "test.servx",
+                "content": {"membership": "join"},
+                "auth_events": [],
+                "prev_state": [(most_recent, {})],
+                "prev_events": [(most_recent, {})],
+                "unsigned": {"malicious garbage": "hackz", "more warez": "more hackz"},
+            }
+        )
+        self.get_success(federation_event_handler.on_receive_pdu("test.serv", event1))
+
+        event = self.get_success(self.store.get_event("$event1:test.serv"))
+        event_dict = event.get_dict()
+        # Make sure unauthorized fields are stripped from unsigned
+        self.assertNotIn("hackz", event_dict["unsigned"])
```

Comment:
I think this assertion will pass if you reverted your code changes. The same applies to line +348 below.

It's going to assert `hacks not in event_dict["unsigned"]`. But `x in my_dict` tests that a _key_ `x` is contained in a dictionary:

```
>>> 1 in {"a": 1}
False
>>> "a" in {"a": 1}
True
```

Reply:
Got it thanks!